### ISSUE TITLE
IFC Export setting - low file size

### ISSUE THREAD

**Issue by aroestia** [Autodesk/revit-ifc#554]
Hi,
what is the recommended setting in order to get the lowest file size for IFC2X3?
where can I find the explanation for the difference between the MVD and how its impact the IFC file (coordination view, basic FM, GSA etc)?

Thanks

**Comment by Teun1** [comment]
If you want a correct IFC just use the built in presets from your certified software.
If you want an IFC that has the lowest file size. Just skip everything up to the point that somebody is complaining. 

There are occasions where people want an IFC -no matter what- but the exporter is not able to handle the model. Could be a problem on either side. In that case we use an automated preset that we call ‘small’ (the famous 1 button option). Its goal is a minimal and balanced geometry. No annotation, minimal IfcSpace, minimal properties. With the following settings based on a 3D view:

3D View: {3D - myview}
   0:00:03.353000
   02-12-2022
   Dynamo exporter version_Q:  23-11-2022
   Settings are based on  =  small
###   GENERAL
   FileVersion = IFC4 (so not IFC2x3)
   IFCFileType   2 (=ifczip obiviously – some viewers do not like this > simply unzip the file)
   ExchangeRequirement   0 (= Architecture)
   SpaceBoundaryLevel = 1 (=1st Level - If rooms are exported make them a little bit usefull)
   SitePlacement   2 (=Survey Point)
   WallAndColumnSplitting = False
   IncludeSteelElements   true
###   ADDITIONAL CONTENT
   ExportAnnotations   false
   ExportLinkedFiles   false
   ExportVisibleElementsInView   true
   ExportRoomsInView   true (or do not export at all of course)
###   PROPERTY SETS
   ExportInternalRevitPropertySets   false
   ExportIFCCommonPropertySets   true (I still want ‘some’ relevant info)
   ExportBaseQuantities = False
   ExportSchedulesAsPsets   false
   ExportSpecificSchedules   false
   ExportUserDefinedPsets   false
   ExportUserDefinedPsetsFileName   :\REVIT\SETTINGS\blabla.txt
   ExportUserDefinedParameterMapping   false
   ExportUserDefinedParameterMappingFileName   :\REVIT\SETTINGS\blabla2.txt
###   LEVEL OF DETAIL
   TessellationLevelOfDetail   0,25 (Extra Low)
###   ADVANCED
   ExportPartsAsBuildingElements   false
   ExportSolidModelRep   true (I let the exporter decide )
   UseActiveViewGeometry   true (get what you see)
   FilterViewId = 11681064
   UseFamilyAndTypeNameForReference   false
   Use2DRoomBoundaryForVolume   true (still rooms without to much problems)
   IncludeSiteElevation   false
   StoreIFCGUID   false
   ExportBoundingBox   false
   UseOnlyTriangulation   true (sometimes more stable result)
   UseTypeNameOnlyForIfcType   false
   UseVisibleRevitNameAsEntityName   false

[Help | IFC Export Setup Options | Autodesk](https://help.autodesk.com/view/RVT/2022/ENU/?guid=GUID-E029E3AD-1639-4446-A935-C9796BC34C95)

**Comment by aroestia** [comment]
Thanks for the replay @Teun1 !

I will be more accurate. 
When I'm saying file size I mean regarding to generation (export) time and the complexity of the geometry. 
I assume (and hope someone will confirm) that the geometry is responsible for the majority of file size and process in export. ?!

I saw that you picked the "ExportSolidModelRep" as true...is that should influence the export processing?
are you familiar with some app/service that will expose me to the segmentation between file size analysis? (geometry VS data parameters).

**Comment by Teun1** [comment]
Leaving things out will effect file size, export or loading time. That is true for properties, objects or the efficiency by wich a geometry can be described.
Just find the sweet spot that is true for your porposes.

And consider also the folowing. Sometimes it is simply better ( or the only way) to generate multiple smaller IFC files than one big file. Because the amount of exported objects is also important. And Revit and or your computer will have difficulty to proces everything at ones.

I am not aware of software to do this kind of analysis.

**Comment by AngelVelezSosa** [comment]
I agree with that synopsis.  We will of course do what we can to improve things over time but clearly the more detail you have, the longer it takes.  And not all detail is created equal.

**Comment by aroestia** [comment]
OK Thnaks, @AngelVelezSosa @Teun1 

2 following questions: 

1. could you please describe a little more about the _**solid model rep**_
2. no matter how big the amount of elements that I trying to export, the export process starts very very fast and becomes slower after 50%-60% of the elements. is it something you are aware of?

Amir

**Comment by AngelVelezSosa** [comment]
1. The "solid model rep" is a hybrid approach in IFC that allows for a mixture of extrusions and faceted BReps, but isn't officially supported in any MVD.  The most obvious example is the default Revit desk - a normal IFC export creates 18 faceted BReps, but the "solid model rep" creates 17 extrusions and 1 faceted BRep, which is better quality and more concise.
2. There are a few possibilities here:
2.a. Later elements take longer to export than earlier elements, because of in-memory structures.  That's (mostly) just life.
2.b. There may be an interesting element at the 50-60% mark that is harder to export.
2.c Sometimes the status screen becomes unresponsive, or the guess for % wrong is off.
2.d. One of many other reasons :)
